UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs

```diff
@@ -98,7 +115,12 @@ public async Task<ExchangeSetServiceResponse> CreateProductDataSinceDateTime(Pro
                 return response;
             }            
 
-            response =await SetExchangeSetResponseLinks(response);
+            var exchangeSetServiceResponse = await SetExchangeSetResponseLinks(response);
+
+            if (!string.IsNullOrEmpty(exchangeSetServiceResponse.BatchId))
+            {
```

Comment:
As discussed we need to handle 304 sales catalogue response. In case of 304 from SCS, ESS will return 200 for post product versions endpoint. In this scenario also message should be saved in queue and json file has to be created with empty Products List so that fulfillment can use it to create empty exchange set.

Reply:
As discussed in call code changes has been for 304 response .